core/app/models/spree/line_item.rb

```diff
@@ -88,12 +90,16 @@ def pre_tax_amount
     alias display_total display_amount
     deprecate display_total: :display_amount, deprecator: Spree::Deprecation
 
-    # Sets price and currency from a `Spree::Money` object
+    # Sets price from a `Spree::Money` object
     #
-    # @param [Spree::Money] money - the money object to obtain price and currency from
+    # @param [Spree::Money] money - the money object to obtain price from
     def money_price=(money)
+      unless money
+        self.price = nil
+        return
+      end
+      raise CurrencyMismatch, "Line item price currency must match order currency!" if money.currency.iso_code != currency
       self.price = money.to_d
```

Comment:
This is a little tough to parse with the early return inside the unless (which has me thinking about the rest of the method like the double negative unless...else).

I think it would read better as an if/elseif/else

``` ruby
  if !money
    self.price = nil
  elsif money.currency.iso_code != currency
    raise CurrencyMismatch, "Line item price currency must match order currency!"
  else
    self.price = money.to_d
  end
```